File: src/Chapter-5-C99/Questions/Question11.c
//Write a program that asks the user for a two-digit number, then prints the
//English word for the number:
//Enter a two-digit number: 45
//You entered the number forty-five.
#include <stdio.h>

int main(void)
{
    int num,first, second; //digit
    printf("Enter a two-digit number: ");
    scanf("%d", &num);
    first = num/10;
    second = num%10;
    if (first == 1)
    {
    if (second == 1)
        {
            printf("Eleven\n");
        }
        else if (second == 2)
        {
            printf("Twelve\n");
        }
        else if (second == 3)
        {
            printf("Thirteen\n");
        }
        else if (second == 4)
        {
            printf("Fourteen\n");
        }
        else if (second == 5)
        {
            printf("Fifteen\n");
        }
        else if (second == 6)
        {
            printf("Sixteen\n");
        }
        else if (second == 7)
        {
            printf("Seventeen\n");
        }
        else if (second == 8)
        {
            printf("Eighteen\n");
        }
        else
        {
            printf("Nineteen\n");
        }
        exit(0);
    }
    switch(first)
    {
    case 2:
        printf("Twenty");
        break;
    case 3:
        printf("Thirty");
        break;
    case 4:
        printf("Forty");
        break;
    case 5:
        printf("Fifty");
        break;
    case 6:
        printf("Sixty");
        break;
    case 7:
        printf("Seventy");
        break;
    case 8:
        printf("Eighty");
        break;
    case 9:
        printf("Ninety");
        break;
    }
    switch(second)
    {
    case 1:
        printf("-one\n");
        break;
    case 2:
        printf("-two\n");
        break;
    case 3:
        printf("-three\n");
        break;
    case 4:
        printf("-four\n");
        break;
    case 5:
        printf("-five\n");
        break;
    case 6:
        printf("-six\n");
        break;
    case 7:
        printf("-seven\n");
        break;
    case 8:
        printf("-eight\n");
        break;
    case 9:
        printf("-nine\n");
        break;
    }

    return 0;
}

File: src/Chapter-5-Rust/Questions/Question11.rs
//Write a program that asks the user for a two-digit number, then prints the
//English word for the number:
//Enter a two-digit number: 45
//You entered the number forty-five.
#[macro_use] extern crate text_io;
//process library to pause console
use std::process::Command;
use std::process;
//Need this to use the print!
use std::io::{self, Write};

fn main()
{
	let (num, first, second) : (i32, i32, i32);
	print!("Enter a two-digit number: ");
	io::stdout().flush().unwrap();
	scan!("{}", num);
	first = num/10;
	second = num%10;
	if first == 1
	{
		match second {
			1 			=> { print!("Eleven\n");}
			2 			=> { print!("Twelve\n");}
			3			=> { print!("Thirteen\n");}
			4 			=> { print!("Fourteen\n");}
			5 			=> { print!("Fifteen\n");}
			6 			=> { print!("Sixteen\n");}
			7 			=> { print!("Seventeen\n");}
			8 			=> { print!("Eighteen\n");}
			_ 			=> { print!("Nineteen\n");}
		}
		io::stdout().flush().unwrap();
		let _ = Command::new("cmd.exe").arg("/c").arg("pause").status();
		process::exit(1);

	}
	//lovely syntax
	match first {
		2 			=> { print!("Twenty");}
		3 			=> { print!("Thirty");}
		4 			=> { print!("Forty");}
		5 			=> { print!("Fifty");}
		6 			=> { print!("Sixty");}
		7           => { print!("Seventy");}
		8 			=> { print!("Eigthy");}
		_ 			=> { print!("Ninety");}
	}
	match second {
		1           => { print!("-one\n");}
		2 			=> { print!("-two\n");}
		3 			=> { print!("-three\n");}
		4 			=> { print!("-four\n");}
		5 			=> { print!("-five\n");}
		6 			=> { print!("-six\n");}
		7           => { print!("-seven\n");}
		8 			=> { print!("-eight\n");}
		_ 			=> { print!("-nine\n");}
	}
	io::stdout().flush().unwrap();
	let _ = Command::new("cmd.exe").arg("/c").arg("pause").status();
}
